### PR TITLE
Pyrsia CLI Configuration Remove Operation

### LINKED ISSUE #1266: Pyrsia CLI Configuration Remove Not Working
### Steps taken/Steps to reproduce

`pyrsia config --remove` not working. Functionality needs to be implemented as it is missing from the existing code base.

use `pyrsia config --show` to get the Pyrsia configuration path. Now use `pyrsia config --remove`. It is expected that configuration file should be cleaned but it still persists. You can confirm using `ls <Config file path>`.

### Expected Behavior

_No response_

### Experienced Instance of the bug

rustc 1.63.0 (4b91a6ea7 2022-08-08)

### PR DESCRIPTION
**Pull request by dgarus**
## Description

Fixes #1266 

Done:
- added `config::config_remove()` function and according unit tests.
- added handler to perform `pyrsia config --remove` operation.
- fixed wrong condition to call `show` operation, before it always was `true`

How to review:
Use `pyrsia config --show` to get the Pyrsia configuration path. Now use `pyrsia config --remove`. 
It is expected that configuration file should be cleaned. You can confirm using ls <Config file path>.

## PR Checklist

- [x] I've read the [contributing guidelines](https://github.com/pyrsia/.github/blob/main/contributing.md).
- [x] I've read ["What is a Good PR?"](https://github.com/pyrsia/pyrsia/blob/main/docs/community/get_involved/good_pr.md)
- [x] I've included a good title and brief description along with how to review them.
- [x] I've linked any associated an [issue](https://github.com/pyrsia/pyrsia/issues).

### Code Contributions

- [x] I've built the code `cargo build --all-targets` successfully.
- [x] I've run the unit tests `cargo test --workspace` and all tests pass.
- [x] I've run the [integrations tests](https://github.com/pyrsia/pyrsia-integration-tests#how-to-set-up-and-run-the-tests) and all tests passes.

(Two tests are failed, but the same tests are failed on `main` too)

### CODE REVIEW

File: src/cli_commands/config.rs [hunk 1]
```diff
@@ -185,35 +193,68 @@ pub fn get_config_file_path() -> Result<PathBuf> {
...
+        });
+    }
+
+    #[test]
+    #[serial]
+    fn test_remove_not_existed_config_file() {
+        config_remove().expect("remove_config failed");
+    }
+
+    fn setup_temp_home_dir_and_execute<F>(op: F)
```
Comment (dasmanas) [inline src/cli_commands/config.rs:244]: Great improvement in terms of reusability. 👍

**Review by dasmanas — APPROVED** [review]
Tested from the forked branch. Remove functionality is now working.
LGTM.

**Review by sbtaylor15 — APPROVED** [review]
/lgtm

File: pyrsia_cli/src/main.rs [hunk 2]
```diff
@@ -57,7 +57,17 @@ async fn main() {
                     }
                 }
             }
-            if config_matches.contains_id("show") {
+            if *config_matches.get_one::<bool>("remove").unwrap() {
```
Comment (fishseabowl) [inline pyrsia_cli/src/main.rs:60]: IMO, tt is better not to use unwrap().

Comment (dgarus) [reply pyrsia_cli/src/main.rs:60]: @fishseabowl we could use `unwrap_or(&false)` here, it would be more reliable. 
What do you think?

Comment (fishseabowl) [reply pyrsia_cli/src/main.rs:60]: @dgarus yes. it is better. Thanks

**Review by fishseabowl — APPROVED** [review]
lgtm

**Review by dasmanas — APPROVED** [review]

**Pull request merged**